### PR TITLE
Add whiptail binary, new libraries, and slang dependency

### PR DESCRIPTION
**Pull request by kylerankin**
The whiptail binary will allow us to create GUI menus from bash scripts.
It is included in the newt library, which depends on slang. To enable,
the board configuration file should add CONFIG_SLANG=y and CONFIG_NEWT=y

### CODE REVIEW

File: modules/newt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+modules-$(CONFIG_TPMTOTP) += newt
```
Comment (flammit) [inline modules/newt]: you meant `modules-$(CONFIG_NEWT)` right?

**Comment by flammit** [comment]
Curious - what's the rom size impact for these additions?

**Comment by kylerankin** [comment]
With whiptail and supporting libs on with my board options:
-rw-r--r-- 1 kyle kyle 2688512 Feb 19 17:17 build/librem13v2/initrd.cpio.xz

Disabling CONFIG_NEWT and CONFIG_SLANG:
-rw-r--r-- 1 kyle kyle 2380288 Feb 19 17:21 build/librem13v2/initrd.cpio.xz

**Comment by osresearch** [comment]
While trying to build `BOARD=qemu-coreboot`, I get an error:

```
make[2]: Entering directory '/home/user/pr/build/slang-2.3.1a'
cd src; /home/user/pr/build/make-4.2/make elf
make[3]: Entering directory '/home/user/pr/build/slang-2.3.1a/src'
/home/user/pr/build/slang-2.3.1a/autoconf/mkinsdir.sh /home/user/pr/build/slang-2.3.1a/src/elfobjs
cp sysconf.h config.h
make[3]: *** No rule to make target '/home/user/pr/build/slang-2.3.1a/src/elfobjs', needed by '/home/user/pr/build/slang-2.3.1a/src/elfobjs/libslang.so.2.3.1'.  Stop.
make[3]: *** Waiting for unfinished jobs....
mkdir -p -- /home/user/pr/build/slang-2.3.1a/src/elfobjs
touch /home/user/pr/build/slang-2.3.1a/src/elfobjs/tstamp
make[3]: Leaving directory '/home/user/pr/build/slang-2.3.1a/src'
make[2]: *** [Makefile:58: elf] Error 2
make[2]: Leaving directory '/home/user/pr/build/slang-2.3.1a'
```

It builds with `BOARD=librem13v2`, so there is some dependency that is not being captured.

**Comment by osresearch** [comment]
This suggests an improvement in our module files: #328

**Pull request merged**